### PR TITLE
 #20413 admin__control-fields class should not add in Frontend ( in checkout page )

### PR DESCRIPTION
**Pull request by MaxRomanov4669**
### Description (*)
fixed issue with admin__control-fields class on checkout

### Fixed Issues
1. https://github.com/magento/magento2/issues/20413: admin__control-fields class should not add in Frontend ( in checkout page )

### Manual testing scenarios (*)
1. Open frontend
2. Add any product in cart
3. Go to checkout and see shipping address form > Street Address Label
4. And inspect html here added admin__control-fields in fieldset
http://prntscr.com/m904ax

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=24786) <br/>All committers have signed the CLA.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi, @MaxRomanov4669. Thanks for collaboration. 
The `group.js` file shouldn't know anything about admin/frontend area. 
In case if it isn't general classes we should pass they through `additionalClasses`, for that, we should add it to XML configuration where we use this component.

**Comment by MaxRomanov4669** [comment]
Hi @VladimirZaets .  Yes, classes 'admin__control-grouped' and 'admin__control-fields' are not general but if we remove these classes from group.js we need to modify a lot of php files like app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/General.php (see customizeNewDateRangeField method) and xml files that contains Magento_Ui/js/form/components/group . Maybe a better way is to create specific group component based on Magento_Ui/js/form/components/group for checkout street address fields? Please advise.

**Comment by VladimirZaets** [comment]
Hi @MaxRomanov4669, yes I know there is impact a lot of configuration but it is a really right way. 
Another option that I don't prefer but it works that introduces another one template (HTML) file, for the frontend area. It will be better than extend JS class just to add additional classes

File: app/code/Magento/Bundle/Ui/DataProvider/Product/Form/Modifier/BundlePanel.php [hunk 1]
```diff
@@ -525,7 +525,8 @@ protected function getOptionInfo()
                         'componentType' => Container::NAME,
                         'component' => 'Magento_Ui/js/form/components/group',
                         'showLabel' => false,
-                        'additionalClasses' => 'admin__field-group-columns admin__control-group-equal',
+                        'additionalClasses' => 'admin__field-group-columns admin__control-group-equal '
```
Comment (VladimirZaets) [inline app/code/Magento/Bundle/Ui/DataProvider/Product/Form/Modifier/BundlePanel.php]: Additional classes can be array, you don't need to concatenate strings

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/AdvancedPricing.php [hunk 2]
```diff
@@ -574,7 +574,7 @@ private function specialPriceDataToInline()
                 $this->meta,
                 [
                     'label' => __('Special Price From'),
-                    'additionalClasses' => 'admin__control-grouped-date',
+                    'additionalClasses' => 'admin__control-grouped-date admin__control-grouped',
```
Comment (VladimirZaets) [inline app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/AdvancedPricing.php:577]: Additional classes can be array, you don't need to concatenate strings

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php [hunk 3]
```diff
@@ -541,7 +541,8 @@ protected function getCommonContainerConfig($sortOrder)
                         'component' => 'Magento_Ui/js/form/components/group',
                         'breakLine' => false,
                         'showLabel' => false,
-                        'additionalClasses' => 'admin__field-group-columns admin__control-group-equal',
+                        'additionalClasses' => 'admin__field-group-columns admin__control-group-equal '
```
Comment (VladimirZaets) [inline app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php]: Additional classes can be array, you don't need to concatenate strings

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php [hunk 4]
```diff
@@ -603,7 +604,8 @@ protected function getStaticTypeContainerConfig($sortOrder)
...
+                            . 'admin__control-grouped',
```
Comment (VladimirZaets) [inline app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php]: Additional classes can be array, you don't need to concatenate strings

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Please take a look at comments
Thanks

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php (hunk 4 above)
Comment (MaxRomanov4669) [reply app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php]: fixed

File: app/code/Magento/Bundle/Ui/DataProvider/Product/Form/Modifier/BundlePanel.php (hunk 1 above)
Comment (MaxRomanov4669) [reply app/code/Magento/Bundle/Ui/DataProvider/Product/Form/Modifier/BundlePanel.php]: fixed

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php (hunk 3 above)
Comment (MaxRomanov4669) [reply app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/CustomOptions.php]: fixed

File: app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/AdvancedPricing.php (hunk 2 above)
Comment (MaxRomanov4669) [reply app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/AdvancedPricing.php:577]: there is no concatenation. Please let me know if I need to use array.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @MaxRomanov4669,
Sorry for long delay for processing your Pull Requst.

Could you resolve conflicts, so we'll be able to review it?

**Comment by ihor-sviziev** [comment]
Hi @MaxRomanov4669,
Will you be able to update your PR?

**Comment by ihor-sviziev** [comment]
@MaxRomanov4669  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**